src/libcharon/plugins/kernel_super/kernel_super_ipsec.c

```diff
@@ -791,6 +903,7 @@ static status_t del_policy_internal(private_kernel_super_ipsec_t* this,
 	switch(del_sp_verdict.verdict) {
 		case DEL_SP_VERDICT_DEL:
 			if (! data->skip_sending_to_terminator) {
+				delete_terminator_tunnels_oper(super_peer_info, picomsg_terminator_conn_get_id(terminator_conn));
```

Comment:
The call to delete_terminator_tunnels_oper()` should happen even if `skip_sending_to_terminator` is TRUE.